External/Plugins/ASCompletion/Completion/ASComplete.cs

```diff
@@ -1257,22 +1257,45 @@ static private bool HandleDeclarationCompletion(ScintillaControl Sci, string tai
             return true;
         }
 
+        /// <summary>
+        /// Lookup type declaration keywords anywhere in the provided text
+        /// </summary>
         private static bool IsTypeDecl(string line, string[] typesKeywords)
         {
+            var max = line.Length - 1;
             foreach (string keyword in typesKeywords)
-                if (line.IndexOf(keyword) >= 0) return true;
+            {
+                var p = line.IndexOf(keyword);
+                if (p >= 0) 
+                {
+                    // verify keyword between spaces
+                    var end = p + keyword.Length;
+                    if ((p == 0 || line[p-1] <= 32) 
+                        && end < max && line[end] <= 32) return true;
+                }
+            }
             return false;
         }
 
+        /// <summary>
+        /// Look if the provided text starts with any declaration keyword
+        /// </summary>
         private static bool IsDeclaration(string line, ContextFeatures features)
         {
             foreach (string keyword in features.accessKeywords)
-                if (line.StartsWith(keyword)) return true;
+                if (line.StartsWith(keyword) && SpaceFollows(line, keyword)) return true;
```

Comment:
To me this "line.StartsWith(keyword) && SpaceFollows(line, keyword)" even if it's used in only two places looks like a total candidate to be converted to an extension method in some other file. And since the logic used in IsTypeDecl is more or less similar, maybe the same could be applied.